src/servers/http_server_v2.cc

```diff
@@ -423,214 +481,224 @@ ReadDataArrayFromJson(
       *base = reinterpret_cast<char*>(&int32_t_tensor[0]);
-    } else if (strcmp(dtype, "INT64")) {
+      break;
+    }
+    case TYPE_INT64: {
       std::vector<int64_t> int64_t_tensor;
-      for (rapidjson::Value::ConstValueIterator itr = tensor_data.Begin();
-           itr != tensor_data.End(); ++itr) {
-        int64_t_tensor.push_back((int64_t)itr->GetInt());
-      }
+      ReadDataFromJsonHelper(&int64_t_tensor, dtype, tensor_data, &counter);
       *base = reinterpret_cast<char*>(&int64_t_tensor[0]);
+      break;
     }
     // FP16 needs a work around
-    else if (strcmp(dtype, "FP16")) {
-      // std::vector<float> float_tensor;
-      // for (rapidjson::Value::ConstValueIterator itr = tensor_data.Begin();
-      //      itr != tensor_data.End(); ++itr) {
-      //   float_tensor.push_back((float)itr->GetFloat());
-      // }
-      // *base = reinterpret_cast<char*>(&float_tensor[0]);
-    } else if (strcmp(dtype, "FP32")) {
+    case TYPE_FP16: {  // std::vector<float> float16_tensor;
+      // ReadDataFromJsonHelper(&float16_t_tensor, dtype, tensor_data,
+      // &counter); *base = reinterpret_cast<char*>(&float16_tensor[0]);
+      break;
+    }
+    case TYPE_FP32: {
       std::vector<float> float_tensor;
-      for (rapidjson::Value::ConstValueIterator itr = tensor_data.Begin();
-           itr != tensor_data.End(); ++itr) {
-        float_tensor.push_back((float)itr->GetFloat());
-      }
+      ReadDataFromJsonHelper(&float_tensor, dtype, tensor_data, &counter);
       *base = reinterpret_cast<char*>(&float_tensor[0]);
-    } else if (strcmp(dtype, "FP64")) {
+      break;
+    }
+    case TYPE_FP64: {
       std::vector<double> double_tensor;
-      for (rapidjson::Value::ConstValueIterator itr = tensor_data.Begin();
-           itr != tensor_data.End(); ++itr) {
-        double_tensor.push_back((double)itr->GetDouble());
-      }
+      ReadDataFromJsonHelper(&double_tensor, dtype, tensor_data, &counter);
       *base = reinterpret_cast<char*>(&double_tensor[0]);
+      break;
     }
     // BYTES (String) needs a work around
-    else if (strcmp(dtype, "BYTES")) {
-    } else {
+    case TYPE_STRING:
+      break;
+    case TYPE_INVALID: {
       return TRTSERVER_ErrorNew(
           TRTSERVER_ERROR_INVALID_ARG,
           std::string(
-              "invalid datatype " + std::string(dtype) + " of input " +
+              "invalid datatype " +
+              std::string(DataTypeToProtocolString(dtype)) + " of input " +
               request_input["name"].GetString())
               .c_str());
     }
-    *byte_size = element_cnt * GetDataTypeByteSize(dtype);
+    default:
+      break;
+  }
+
+  int element_cnt = 0;
+  for (rapidjson::SizeType i = 0; i < shape.Size(); i++) {
+    if (element_cnt == 0) {
+      element_cnt = shape[i].GetInt();
+    } else {
+      element_cnt *= shape[i].GetInt();
+    }
   }
 
+  *byte_size =
+      element_cnt * GetDataTypeByteSize(DataTypeToProtocolString(dtype));
```

Comment:
There's a GetByteSize that takes DataType directly... in model_config_utils I think.